cfme/networks/provider/nuage.py

```diff
@@ -44,6 +46,27 @@ class events(Tab, BeforeFillMixin):
         validate = Button('Validate')
 
 
+class ValidateStatsMixin(object):
+    def validate_stats(self, host_stats, ui=False):
+        """ Validates that the detail page matches the subnet's inventory.
+
+        Args:
+            host_stats: dictionary of values to be compered to UI values,
+            where keys have the same names as NuageSubnet ui functions
+
+        This method gets host stats and matches them against the UI. An exception will
```

Comment:
_matches them against the UI_ - that part is not always true they way it's implemented now. In fact, it's not even true in default case because we set default to
```
ui=False
```
right?

Since we currently only support UI approach and we don't have desire to be checking DB really (trying to simulate what user sees as much as possible) I'd suggest to just drop `ui=False` parameter of our `validate_stats` function and always go for UI method. We can easily support both methods later should need arise.

Reply:
Also: _This method gets host stats and matches them against the UI_ sounds much like the method itself somehow retrieves host stats. Please reword, we want it to be clear that we're passing dictionary of expected values. No need to even name them as "host_stats" but maybe rather "expected_stats".

Sorry for nitpicking, but it pisses me off when I look at other providers how confusing they are. They say "get_stats" and mean PhysicalServer status like "on" or "off". Then they say "validate_stats" and they mean server details. So confusing 😄 

Also, please add a short clarification like _Please make sure NuageSubnet implements counterpart method for each of the keys in `expected_stats` dictionary._